src/lib/swapper/swappers/CowSwapper/cowApprove/cowApprove.ts

```diff
@@ -1,60 +1,42 @@
+import { fromAssetId } from '@shapeshiftoss/caip'
 import type { KnownChainIds } from '@shapeshiftoss/types'
 import type { ApproveAmountInput, ApproveInfiniteInput } from 'lib/swapper/api'
-import { SwapError, SwapErrorType } from 'lib/swapper/api'
 import type { CowSwapperDeps } from 'lib/swapper/swappers/CowSwapper/CowSwapper'
 import { MAX_ALLOWANCE } from 'lib/swapper/swappers/CowSwapper/utils/constants'
-import { erc20Abi } from 'lib/swapper/swappers/utils/abi/erc20-abi'
 import { grantAllowance } from 'lib/swapper/swappers/utils/helpers/helpers'
 
-export async function cowApproveInfinite(
+export function cowApproveInfinite(
   { adapter, web3 }: CowSwapperDeps,
   { quote, wallet }: ApproveInfiniteInput<KnownChainIds.EthereumMainnet>,
-) {
-  try {
-    const allowanceGrantRequired = await grantAllowance<KnownChainIds.EthereumMainnet>({
-      quote: {
-        ...quote,
-        sellAmountBeforeFeesCryptoBaseUnit: MAX_ALLOWANCE,
-      },
-      wallet,
-      adapter,
-      erc20Abi,
-      web3,
-    })
+): Promise<string> {
+  const { accountNumber, allowanceContract, feeData, sellAsset } = quote
 
-    return allowanceGrantRequired
-  } catch (e) {
-    if (e instanceof SwapError) throw e
-    throw new SwapError('[cowApproveInfinite]', {
-      cause: e,
-      code: SwapErrorType.APPROVE_INFINITE_FAILED,
-    })
-  }
+  return grantAllowance({
+    accountNumber,
+    approvalAmount: MAX_ALLOWANCE,
```

Comment:
Obviously not for this PR but would be nice to make a discriminated union of `approvalAmount` / `infinite: boolean` (which doesn't exist yet)

Reply:
Simplified it a bit more, not sure if discrimination is worth